gtime/causality/granger_causality.py

```diff
@@ -0,0 +1,213 @@
+import numpy as np
+import pandas as pd
+
+from scipy import stats
+from sklearn.base import BaseEstimator
+from sklearn.linear_model import LinearRegression
+from sklearn.metrics import mean_squared_error
+
+class GrangerCausality(BaseEstimator):
+    """Class to check for Granger causality between two time series, i.e. 
+    to check if a time series X causes Y: X->Y.
+
+    Parameters
+    ----------
+    target_col : str, 
+        The column to use as the reference, i.e. the time series Y
+    x_col : str
+        The column to test for Granger causality, i.e. the time 
+        series X.
+    max_shift : int
+        The maximal number of shifts to check for Granger causality
+    statistics : str, optional, default: 'ssr_F'
+        The statistical test to perform for Granger causality. Either 'ssr_F'
+        (sum squared residuals with F-test), 'ssr_chi2' (sum squared residuals 
+        with chi square test), 'likelihood_chi2' (likelihood ratio test with 
+        chi square distribution), 'zero_F' (F-test that all lag coefficients of 
+        the time series X are zero) or 'all' (perform all the tests mentioned 
+        above)
+
+    Examples
+    --------
+    >>> from gtime.causality.granger_causality import GrangerCausality
+    >>> import pandas.util.testing as testing
+    >>> data = testing.makeTimeDataFrame(freq="s", nper=1000)
+    >>> GrangerCausality(target_col='A', x_col='B', max_shift=10, statistics='ssr_F').fit(data)
+                        ssr F-test
+    F-value               0.372640
+    p-value               0.958527
+    degrees of freedom  969.000000
+    number of shifts     10.000000
+    """
+    def __init__(self, 
+                 target_col: str, 
+                 x_col: str, 
+                 max_shift: int, 
+                 statistics='ssr_F'):
```

Comment:
Maybe taking a list of string as input instead ? What do you think ?

Reply:
I would personally stick with multiple parameters as it is now, I think is more clearer this way